packages/flutter/test/widgets/animated_size_test.dart

```diff
@@ -88,6 +102,57 @@ void main() {
       expect(box.size.height, equals(100.0));
     });
 
+    testWidgetsWithLeakTracking('calls onEnd when animation is completed', (WidgetTester tester) async {
+      await tester.pumpWidget(
```

Comment:
You can simplify this. remove the `mockOnEndFunction` stuff and just define an inline function here:

```dart
int callCount = 0;
void handleEnd() {
  callCount++;
}
```

and then just pass `handleEnd` wherever you need it

Reply:
Done